src/fheroes2/heroes/heroes_action.cpp

```diff
@@ -3295,12 +3298,21 @@ namespace
                     I.getRadar().SetRenderArea( eyeRoi );
                     I.redraw( Interface::REDRAW_GAMEAREA | Interface::REDRAW_RADAR );
 
+                    if ( skipAnimation ) {
+                        continue;
+                    }
```

Comment:
If we skip viewing the rest "Eyes" then we need only to redraw the radar part without gamearea redraw. Gamearea will be rendered after we finish this loop at line 3327.
So we can move `I.getGameArea().SetCenter( eyePosition );` (from line 3294) after the check for `skipAnimation` and if `skipAnimation` is `true` do only `REDRAW_RADAR`:

```suggestion
                    if ( skipAnimation ) {
                        I.redraw( Interface::REDRAW_RADAR );
                        continue;
                    }

                    I.getGameArea().SetCenter( eyePosition );
                    I.redraw( Interface::REDRAW_GAMEAREA | Interface::REDRAW_RADAR );
```